server/src/main/java/org/elasticsearch/common/bytes/BytesReference.java

```diff
@@ -118,6 +118,18 @@ static BytesReference fromByteBuffer(ByteBuffer buffer) {
      */
     byte get(int index);
 
+    /**
+     * Bulk get operation that allows reading a specified number of bytes from a specified index in this instance to
+     * a specified index in the given byte array.
+     *
+     * @param buf    array to read bytes into
+     * @param index  index in this instance to read from
+     * @param offset offset in the array to read first byte into
+     * @param length number of bytes to read
+     * @return number of bytes that were actually copied to the buffer
+     */
+    int get(byte[] buf, int index, int offset, int length);
```

Comment:
Slight preference for a name other than `get`, ideally involving the word `copy`, and also for collecting the parameters `buf/offset/length` together rather than putting `index` in the middle like that.

```suggestion
    int copyTo(int index, byte[] buf, int offset, int length);
```

Reply:
I'm not so sure about that, seems a little non-standard. Both `io.netty.buffer.ByteBuf#getBytes(int, byte[], int, int)` and `java.nio.ByteBuffer#get(int, byte[], int, int)` use this order of parameters so maybe so should we?

Reply:
I'm confused, both the examples you give have the same order of parameters as what I'm suggesting here?

Ack, however, that `get` seems to be a standard name. I don't like it, but it's not a strong opinion.

Reply:
:facepalm: nevermind I had to read it like 5 times to see that the order of parameters is different. Sorry for the noise, reordered the params now.